torchtune/training/checkpointing/_checkpoint_client.py

```diff
@@ -365,6 +408,9 @@ def save_checkpoint(
         adapter_config: Optional[dict[str, Any]] = None,
         adapter_only: bool = False,
         single_device: bool = False,
+        *,
+        full_tensors: bool = True,
```

Comment:
Maybe add a comment in the docstring explaining what full_tensors are and when you might want to use them.